sui_core/src/authority/authority_store.rs

```diff
@@ -710,40 +622,52 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
                 .map(|(object_id, (_, new_object))| (object_id, new_object)),
         )?;
 
-        // Update the indexes of the objects written
-
-        // This is the critical region: testing the locks and writing the
-        // new locks must be atomic, and no writes should happen in between.
-        {
-            // Acquire the lock to ensure no one else writes when we are in here.
-            let _mutexes = self.acquire_locks(&active_inputs[..]);
```

Comment:
Why did we have this critical region before? And why is it safe to break it?

Reply:
I see @gdanezis has a longer comment below, so I'll just respond to that one